lib/mergeBuffers.js

```diff
@@ -25,15 +25,24 @@ function mergeBuffers(gltf) {
         var bufferViewIds = bufferViewsForBuffers[bufferId];
         var length = bufferViewIds.length;
         for (var i = 0; i < length; ++i) {
+            // Check if new bufferView needs to be aligned to 4 bytes.
+            var alignment = lengthSoFar & 3;
+            if (alignment > 0) {
+                var bytesToPad = 4 - alignment;
+                var emptyBuffer = Buffer.alloc(bytesToPad);
+                buffersToMerge.push(emptyBuffer);
+                lengthSoFar += bytesToPad;
+            }
+
+            // Get the data from the bufferView.
             var bufferView = bufferViews[bufferViewIds[i]];
-            bufferView.byteOffset += lengthSoFar;
+            var sourceBufferViewData = buffer.extras._pipeline.source.slice(bufferView.byteOffset, bufferView.byteOffset + bufferView.byteLength);
+            bufferView.byteOffset = lengthSoFar;
             bufferView.buffer = 0;
-        }
 
-        var sourceBuffer = getBufferPadded(buffer.extras._pipeline.source);
-        var sourceBufferLength = sourceBuffer.length;
-        buffersToMerge.push(sourceBuffer);
-        lengthSoFar += sourceBufferLength;
+            buffersToMerge.push(sourceBufferViewData);
+            lengthSoFar += sourceBufferViewData.length;
+        }

```

Comment:
Cool, I like the approach. It could even be simplified more now by just iterating over the buffer views directly. I don't think there's any need for the `getBufferViewsForBuffers` stuff now.

Reply:
Done.